Source/WebCore/platform/graphics/gstreamer/eme/CDMOpenCDM.cpp

```diff
@@ -290,38 +313,35 @@ void CDMInstanceOpenCDM::requestLicense(LicenseType licenseType, const AtomicStr
 void CDMInstanceOpenCDM::updateLicense(const String& sessionId, LicenseType, const SharedBuffer& response, LicenseUpdateCallback callback)
 {
     std::string responseMessage;
-    // FIXME: This is a realy awful API, why not just take a string& ?
-    int haveMessage = m_openCdmSession->Update(reinterpret_cast<unsigned char*>(const_cast<char*>(response.data())), response.size(), responseMessage);
-    GST_DEBUG("session id %s, calling callback %s message", sessionId.utf8().data(), haveMessage ? "with" : "without");
-
-    if (haveMessage) {
-        GST_TRACE("OpenCDM::Update has a message for us");
+    media::OpenCdm::KeyStatus keyStatus = m_openCdmSession->Update(reinterpret_cast<unsigned char*>(const_cast<char*>(response.data())), response.size(), responseMessage);
```

Comment:
Ideally we'll want this to be a `CDMInstance::KeyStatusVector`, but that will require serialisation (perhaps using JSON) to communicate sanely from the CDM.

Reply:
I don't know if I understand you correctly but I think this would violate layers, right?